mailpoet/assets/js/src/public.jsx

```diff
@@ -85,12 +155,31 @@ jQuery(($) => {
       return;
     }
     const recaptcha = $(element);
+    const form = $(recaptcha).parent('form');
     const sitekey = recaptcha.attr('data-sitekey');
+    let size = recaptcha.attr('data-size');
+
+    // Users should not be able to change the size if it is equal to 'invisible' as this would
+    // change the type of the ReCaptcha.
+    if (size !== 'invisible') {
+      size = Hooks.applyFilters('mailpoet_re_captcha_size', 'compact');
+    }
+
     const container = recaptcha.find('> .mailpoet_recaptcha_container').get(0);
     const field = recaptcha.find('> .mailpoet_recaptcha_field');
     if (sitekey) {
-      const size = Hooks.applyFilters('mailpoet_re_captcha_size', 'compact');
-      const widgetId = window.grecaptcha.render(container, { sitekey, size });
+      const params = { sitekey, size };
+
+      if (size === 'invisible') {
+        params.callback = function invisibleReCaptchaCallback(token) {
+          const formData = form.mailpoetSerializeObject() || {};
+          formData.data.recaptcha = token;
```

Comment:
From what I understand, this function `invisibleReCaptchaCallback` will be executed by the recaptcha API as a callback. 
Would these variables `form`, `formData` be available within the function scope?.

I can see this error in the console
<img width="745" alt="Screenshot 2022-04-21 at 1 10 23 PM" src="https://user-images.githubusercontent.com/30554163/164455279-3377bfe5-abe6-4c90-ac6b-1e46a75320cf.png">

When using the Invisible ReCaptcha option, submitting the form and solving the challenge.

Reply:
> From what I understand, this function invisibleReCaptchaCallback will be executed by the recaptcha API as a callback.

That is correct.

>  Would these variables form, formData be available within the function scope?.

I believe you are referring only to `form` as `formData` is defined inside the callback function. I'm not a JS specialist so I could be missing something here, but I believe the answer is that `form` is available within the callback function because of the way that local scope works in JS. A variable defined inside a function is only accessible from that function and any nested functions.

> I can see this error in the console

I don't see this error in my environment. I made some changes related to another problem that you mentioned in the review comment. Could you please test again and check if you still get it?